packages/material-ui/src/Tooltip/Tooltip.js

```diff
@@ -105,86 +145,96 @@ class Tooltip extends React.Component {
     // Fallback to this default id when possible.
     // Use the random value for client side rendering only.
     // We can't use it server-side.
-    this.defaultId = `mui-tooltip-${Math.round(Math.random() * 1e5)}`;
+    defaultId.current = `mui-tooltip-${Math.round(Math.random() * 1e5)}`;
```

Comment:
I think we can put this into a separate effect. Or do we actually need to create a different one if the title changes?

Reply:
I was wondering about the same. I guess it doesn't matter?

Reply:
This should check if we already set it. Otherwise it will change the ID every time we display the Tooltip.